sources/core/Stride.Core/Diagnostics/ProfilingKey.cs

```diff
@@ -39,20 +44,13 @@ public ProfilingKey([NotNull] string name, ProfilingKeyFlags flags = ProfilingKe
         /// <param name="name">The name.</param>
         /// <exception cref="System.ArgumentNullException">parent</exception>
         public ProfilingKey([NotNull] ProfilingKey parent, [NotNull] string name, ProfilingKeyFlags flags = ProfilingKeyFlags.None)
+            : this($"{parent ?? throw new ArgumentNullException(nameof(parent))}.{ValidateNameNotEmpty(name)}", flags)
         {
-            if (parent == null) throw new ArgumentNullException(nameof(parent));
-            if (name == null) throw new ArgumentNullException(nameof(name));
-            Children = new List<ProfilingKey>();
```

Comment:
I know it's tempting to have a one liner, but it's just not very readable I don't think.
I would rather you send over `$"{parent}.{name}"` regardless of how valid those two values are, and throw in this constructor. In both cases it will throw from this constructor anyway.
Is there a particular reason or use case why `parent` can be whitespace but not `name` ? You don't check parent but you do check name for whitespace.

Reply:
Oh, right, parent is a `ProfilingKey`, my bad